lib/pure/uri.nim

```diff
@@ -21,6 +22,50 @@ type
 {.deprecated: [TUrl: Url, TUri: Uri].}
 
 {.push warning[deprecated]: off.}
+
+proc encodeUrl*(s: string): string =
+  ## Encodes a value to be HTTP safe: This means that characters in the set
+  ## ``{'A'..'Z', 'a'..'z', '0'..'9', '_'}`` are carried over to the result,
+  ## a space is converted to ``'+'`` and every other character is encoded as
+  ## ``'%xx'`` where ``xx`` denotes its hexadecimal value.
+  result = newStringOfCap(s.len + s.len shr 2) # assume 12% non-alnum-chars
+  for i in 0..s.len-1:
+    case s[i]
+    of 'a'..'z', 'A'..'Z', '0'..'9', '_': add(result, s[i])
+    of ' ': add(result, '+')
+    else:
+      add(result, '%')
+      add(result, toHex(ord(s[i]), 2))
+
+proc handleHexChar*(c: char, x: var int) {.inline.} =
```

Comment:
Could this procedure be used instead? https://nim-lang.org/docs/parseutils.html#parseHex,string,int,int,int